src/os/bluestore/BlueStore.cc

```diff
@@ -2878,6 +2878,19 @@ void BlueStore::Blob::split(Collection *coll, uint32_t blob_offset, Blob *r)
 	   << "    and " << *r << dendl;
 }
 
+
+void BlueStore::Blob::maybe_prune_tail() {
+  if (get_blob().is_shared()) {
+    return;
+  }
+
+  if (!get_blob().is_compressed() && get_blob().can_prune_tail()) {
```

Comment:
you might want to move !is_shared() && !is_compressed() check into can_prune_tail() and simplify the code here